### ISSUE TITLE
Enums treated as structural equivalent although they aren't

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#11083]
**TypeScript Version:**  2.0.3

**Code**

File: api.d.ts 

``` ts
export enum SymbolKind {
    A = 0,
    B = 999
}
```

File: internal.ts

``` ts
export enum SymbolKind {
    A,
    B
}
```

File: main.ts

``` ts
import * as api from './api';
import * as internal from './internal';

let s1: api.SymbolKind;
let s2: internal.SymbolKind;

s1 = s2;
```

This should produce an error.

The reason why we think this should produce an error is as follows: if you change the api.d.ts to 

``` ts
export enum SymbolKind {
    A,
    B
}
```

You get a compile error. The only reasonable explanation we could come up with was that in the case of a d.ts file the compiler can't make any assumption about the real enum values since the implementing ts file could have assigned some. So this is a real problem. But flagging this as a problem should flag not matching enum values as a problem as well.

**Comment by sandersn** [comment]
Unfortunately, this is the way that "semi-structural" assignability of enums was intended to work when we added it to 1.8. We don't check the initialisers of the enum, just the member names.

In 2.0 we made it slightly stricter, so that computed enums (those with non-constant initialisers) are not assignable to union enums (those with constant or no initialisers). But that doesn't apply here.

We could make it stricter still by checking the types of individual enum members. If they are constant, they'll be a literal number type. But this would be a breaking change, and one that's more likely to affect people than the computed/union break in 2.0. Maybe we should revisit this decision.
